Sprint 04

Larexander opened the pull request.

Larexander committed:
Установлены шрифты, добавлены цвета из фигмы, ЛаунчСкрин подогнан под разные устройства.

Larexander committed:
The application screen is folded according to the layout and covers most cases.

Larexander committed:
Interface elements are linked to the code, the controller lifecycle is written.

Larexander committed:
Fixed image.

ABashkirova commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,10 +1,170 @@
+            title: result.title,
+            message: result.text,
+            preferredStyle: .alert)
+        let action = UIAlertAction(title: result.buttonText, style: .default) {_ in
```
после фигурной скобки в данном случае нужен пробел
```suggestion
        let action = UIAlertAction(title: result.buttonText, style: .default) { _ in
```

ABashkirova commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,10 +1,170 @@
+            self.noButton.isEnabled = true
+            self.yesButton.isEnabled = true
+        }
+        imageView.layer.masksToBounds = true
+        imageView.layer.borderWidth = 8
+        imageView.layer.cornerRadius = 20
+        imageView.layer.borderColor = isCorrect ? UIColor.green.cgColor : UIColor.red.cgColor
```
masksToBounds, cornerRadius  – настройки, которые можно сделать один раз на сториборде или во viewDidLoad

ABashkirova replied:
Для цвета из ассетов есть инициализатор UIColor(named: ..)

ABashkirova commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,10 +1,170 @@
+        self.yesButton.isEnabled = false
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) {
+            self.showNextQuestionOrResults()
+            self.noButton.isEnabled = true
+            self.yesButton.isEnabled = true
```
🙌

Larexander committed:
1. The radius of the poster corners is now 20

2. The background color is black, the color of the poster frames corresponds to the layout

3. The fields of the main stack are 20-20, 10-0

4. Font sizes match the layout

5. Minor fixes

Larexander committed:
Fixed fonts

ABashkirova commented on MovieQuiz/Presentation/Base.lproj/Main.storyboard:
```diff
@@ -1,27 +1,143 @@
+                        <rect key="frame" x="0.0" y="0.0" width="430" height="932"/>
                         <autoresizingMask key="autoresizingMask" widthSizable="YES" heightSizable="YES"/>
+                        <subviews>
+                            <stackView opaque="NO" contentMode="scaleToFill" axis="vertical" spacing="8" translatesAutoresizingMaskIntoConstraints="NO" id="G6s-8d-rLk" userLabel="Main Stcak">
```
по дизайну spacing 20

ABashkirova commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,10 +1,168 @@
+            self.yesButton.isEnabled = true
+        }
+        imageView.layer.borderWidth = 8
+        imageView.layer.borderColor = isCorrect ? UIColor(named: "YP Green (iOS)")!.cgColor : UIColor(named: "YP Red (iOS)")!.cgColor
```
Я не рекомендую использовать focreunwrapping в учебном коде (даже если это работает), это не самая лучшая черта работ будущего джуна
```suggestion
        imageView.layer.borderColor = isCorrect ? UIColor(named: "YP Green (iOS)")?.cgColor : UIColor(named: "YP Red (iOS)")?.cgColor
```

однако как константу (в расширение Colors в helper), можешь вынести статическим свойством с force unwrap'ом – для констант это в целом приемлемо

ABashkirova commented on MovieQuiz/Resources/Info.plist:
```diff
@@ -4,6 +4,15 @@
 	<dict>
+        <key>UIAppFonts</key>
+             <array>
+                 <string>YS Display-Bold</string>
+                 <string>YS Display-Heavy</string>
+                 <string>YS Display-Light</string>
+                 <string>YS Display-Medium</string>
+                 <string>YS Display-Regular</string>
+                 <string>YS Display-Thin</string>
```
лучше оставить только необходимые

Larexander committed:
Fixed the distance between stack elements

Larexander committed:
Minor fixes

Larexander commented:
The fourth sprint has been successfully completed
